### PR TITLE
Created a new class CommandGenerator in command_generator.py 

### PR DESCRIPTION
**Pull request by Yaswanth14**
Worked on issue #4  
Created a new class CommandGenerator in command_generator.py and moved to generate_* functions from main to the new class

### CODE REVIEW

**Comment by sampathbalivada** [comment]
Hi @Yaswanth14, could you please share a screenshot of successful run of the script?

**Comment by Yaswanth14** [comment]
Hey @sampathbalivada, I observed few errors while running the script which are resolved now

### Screenshot of successful run of script:

image

### Details of the updated commit:

- Creating an instance for `get_command_template` as `command_template` in command_generator.py
- Redefining the instaces created for `generate_*` methods

File: src/command_generator.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+class CommandGenerator:
+
+    def __init__(self) -> None:
+        serialize = serializer.JSONSerializer()
+        self.commands = []
+        self.command_template = serialize.get_command_template()
+
+
+    def add_command(self, command):
```
Comment (sampathbalivada) [inline src/command_generator.py]: I don't see where we are using the method. Remove this method.

File: src/main.py [hunk 2]
```diff
@@ -22,6 +24,14 @@ def __init__(self) -> None:
         self.end_commands = serialize.get_end_commands()
         self.config = serialize.get_config()
 
+        self.command_generator = command_generator.CommandGenerator()
+        # Create instances of the generate_* methods
```
Comment (sampathbalivada) [inline src/main.py]: Instead of copying the method reference, modify the code in other places to use the `self.command_generator` instance.

**Review by sampathbalivada — CHANGES_REQUESTED** [review]
Please modify as specified.

**Comment by Yaswanth14** [comment]
> Please modify as specified.

Modified the code, kindly review the changes made

**Review by sampathbalivada — APPROVED** [review]
LGTM. Please share a screenshot of the sample run so that I can merge the PR.

**Comment by Yaswanth14** [comment]
> LGTM. Please share a screenshot of the sample run so that I can merge the PR.

### Here's the screenshot of sample run
image

**Pull request closed without merging**